packages/twenty-server/src/engine/api/rest/api-rest-query-builder/utils/map-field-metadata-to-graphql-query.utils.ts

```diff
@@ -3,6 +3,7 @@ import { RelationMetadataType } from 'src/engine/metadata-modules/relation-metad
 
 const DEFAULT_DEPTH_VALUE = 2;
 
+// TODO: Should be properly type and based on composite type definitions
 export const mapFieldMetadataToGraphqlQuery = (
```

Comment:
@martmull Some refactor can be done here too, maybe later 🙂
nit: a lot of functions are not typed in the api-rest module

Reply:
Yes, this is basically copied from the front `useGenerate<QueryType>RecordQuery.ts`. It would indeed be great to have a dedicated package to factorize this functions accros packages